tasks/main.yml

```diff
@@ -138,13 +140,15 @@
          prometheus_alert_rules_additional.changed }}
     restart_policy: always
     state: started
+    user: "{{ prometheus_docker_user | default(omit) }}"
     volumes:
       # We could mount /etc/prometheus instead but the Docker image contains
       # additional (optional) files
       - /etc/prometheus/rules/:/etc/prometheus/rules/:ro
       - /etc/prometheus/prometheus.yml:/etc/prometheus/prometheus.yml:ro
       - /etc/prometheus/targets/:/etc/prometheus/targets/:ro
-      - prometheus-data:/prometheus
+      - "{{ prometheus_docker_data_volume }}:/prometheus"
```

Comment:
In theory they could, but that's not something we can easily fix at the role level without a lot more complexity. It's also a problem for any of our roles that assume write access by a process.

This is one example where K8s is better, since in a Helm chart you can easily namespace everything.